qiskit_machine_learning/algorithms/inference/qbayesian.py

```diff
@@ -0,0 +1,377 @@
+            qc.measure_all()
+            # Run circuit
+            self._samples = self._run_circuit(qc)
+        else:
+            # Get Grover operator if evidence not empty
+            grover_op = self._get_grover_op(evidence)
+            # Amplitude amplification
+            true_e = {(self._label2qubit[e_key], e_val) for e_key, e_val in evidence.items()}
+            meas_e: Set[Tuple[str, int]] = set()
+            best_qc, best_inter = QuantumCircuit(), 0
+            self._converged = False
+            k = -1
+            # If the measurement of the evidence qubits matches the evidence stop
+            while (true_e != meas_e) and (k < self._limit):
+                # Increment power
+                k += 1
+                # Create circuit with 2^k times Grover operator
+                qc, meas_e = self.__power_grover(grover_op=grover_op, evidence=evidence, k=k)
+                # Test number of
+                if len(true_e.intersection(meas_e)) > best_inter:
+                    best_qc = qc
+            if true_e == meas_e:
+                self._converged = True
+
+            # Create a classical register with the size of the evidence
+            best_qc_meas = QuantumCircuit(*self._circ.qregs)
+            best_qc_meas.append(best_qc, self._circ.qregs)
+            measurement_qcr = ClassicalRegister(self._circ.num_qubits - len(evidence))
+            best_qc_meas.add_register(measurement_qcr)
+            # Map the query qubits to the classical bits and measure them
+            query_qubits = [
+                (label, self._label2qidx[label], qubit)
+                for label, qubit in self._label2qubit.items()
+                if label not in evidence
+            ]
+            query_qubits_sorted = sorted(query_qubits, key=lambda x: x[1], reverse=True)
+            # Measure query variables and return their count
+            best_qc_meas.measure([q[2] for q in query_qubits_sorted], measurement_qcr)
+            # Run circuit
+            counts = self._run_circuit(best_qc_meas)
+            # Build default string with evidence
+            query_string = ""
+            var_idx_sorted = [
+                label for label, _ in sorted(self._label2qidx.items(), key=lambda x: x[1])
+            ]
+            for var in var_idx_sorted:
+                if var in evidence:
+                    query_string += str(evidence[var])
+                else:
+                    query_string += "q"
+            # Retrieve valid samples
+            self._samples = {}
+            # Replace placeholder q with query variables from samples
+            for key, val in counts.items():
+                query = query_string
+                for char in key:
+                    query = query.replace("q", char, 1)
+                self._samples[query] = val
+        if not format_res:
+            return self._samples
+        else:
+            return self._format_samples(self._samples, list(evidence.keys()))
```

Comment:
In the case of formatted a dict is returned that if the user modifies it in anyway, as part of their code, it will not affect later inference. However the self._samples is directly the dict that could be used in later interference (same as returned by the property getter below - which I guess will be an empty dict if this method has not yet been called. One could return a copy of the dict - maybe allowing a user to alter it allows advance use. Maybe we should say something that changing the unformatted dict returned by this (or the one from getter) can affect any later inference?

Reply:
Makes sense. I will return a deep copy.